packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2849,4 +2863,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.inAbstractClass = oldInAbstractClass;
       }
     }
+
+    tsIsAccessModifier(modifier: string): boolean {
+      return (
+        modifier === "private" ||
+        modifier === "public" ||
+        modifier === "protected"
+      );
```

Comment:
Consider extract this function outside of the class, (it's static anyway) so we can use [predicate functions](https://flow.org/en/docs/types/functions/#toc-predicate-functions):
```flow
function tsIsAccessModifier(modifier: string): boolean %checks {
  return (
    modifier === "private" || modifier === "public" || modifier === "protected"
  );
}
```
In this way Flow will infer that `modifier` must be an `Accessibility` enum when `tsIsAccessModifier(modifier)` is `true`.

Reply:
I didn't know how to define predicate function with Flow, Thanks!